flink-runtime-web/web-dashboard/src/app/pages/job/exceptions/job-exceptions.component.ts

```diff
@@ -31,28 +31,45 @@ import { JobService } from 'services';
 export class JobExceptionsComponent implements OnInit {
   rootException = '';
   listOfException: JobExceptionItemInterface[] = [];
+  truncated = false;
+  isLoading = false;
+  maxExceptions = 0;
 
   trackExceptionBy(_: number, node: JobExceptionItemInterface) {
     return node.timestamp;
   }
-
-  constructor(private jobService: JobService, private cdr: ChangeDetectorRef) {}
-
-  ngOnInit() {
+  loadMore() {
+    this.isLoading = true;
+    this.maxExceptions += 10;
     this.jobService.jobDetail$
       .pipe(
         distinctUntilChanged((pre, next) => pre.jid === next.jid),
-        flatMap(job => this.jobService.loadExceptions(job.jid))
+        flatMap(job => this.jobService.loadExceptions(job.jid, this.maxExceptions))
       )
-      .subscribe(data => {
-        // @ts-ignore
-        if (data['root-exception']) {
-          this.rootException = formatDate(data.timestamp, 'yyyy-MM-dd HH:mm:ss', 'en') + '\n' + data['root-exception'];
-        } else {
-          this.rootException = 'No Root Exception';
+      .subscribe(
+        data => {
+          // @ts-ignore
+          if (data['root-exception']) {
+            this.rootException =
+              formatDate(data.timestamp, 'yyyy-MM-dd HH:mm:ss', 'en') + '\n' + data['root-exception'];
+          } else {
+            this.rootException = 'No Root Exception';
+          }
+          this.truncated = data.truncated;
+          this.listOfException = data['all-exceptions'];
+          this.isLoading = false;
```

Comment:
is it worth moving the isLoading  = false, and the cdr.markForCheck() into `finalize` from rxjs to avoid the repetition?

Looks good otherwise.

Reply:
thanks for your advice, use `tap` instead of `finalize` since the source never complete.

Reply:
Makes sense. Thanks!